ts/customEvents.ts

```diff
@@ -0,0 +1,253 @@
+import type Highcharts from 'highcharts';
+import type { Chart, YAxisOptions } from 'highcharts';
+
+/**
+* Custom events v4.0.0 (2025-07-30)
+*
+* (c) 2012-2025 Black Label
+*
+* License: Creative Commons Attribution (CC)
+*/
+
+/**
+ * @namespace customEvents
+ * @description Plugin that adds custom event handling to Highcharts elements
+ */
+
+/**
+ * Highcharts Custom Events Plugin
+ * 
+ * This plugin extends Highcharts to support custom events on various chart elements
+ * including titles, axis elements, plot lines/bands, and data labels.
+ * 
+ * @param {typeof Highcharts} H - The Highcharts library instance
+ * @returns {void}
+ * 
+ * @example
+ * ```javascript
+ * import ObjectEventsPlugin from './customEvents';
+ * ObjectEventsPlugin(Highcharts);
+ * ```
+ */
+export default function ObjectEventsPlugin(H: typeof Highcharts) {
+	// This is a global flag to prevent the plugin from being loaded more than once
+	if (H.customEventsPluginLoaded) {
+		return;
+	} else { 
+		H.customEventsPluginLoaded = true;
+	}
+
+	/**
+	 * Binds DOM events to a Highcharts SVGElement safely,
+	 * avoiding duplicate bindings and tracking for cleanup.
+	 *
+	 * @param {Highcharts.SVGElement} el - The Highcharts SVGElement to bind events to
+	 * @param {Highcharts.ElementEvents} [handlers] - Object mapping event names to callback functions
+	 * @param {Highcharts.BoundEvent[]} boundEvents - Array to track bound events for cleanup
+	 * @returns {void}
+	 * 
+	 * @example
+	 * ```javascript
+	 * bindElementEvents(chart.title, {
+	 *   click: function(e) { console.log('Title clicked!'); },
+	 *   mouseover: function(e) { console.log('Title hovered!'); }
+	 * }, boundEvents);
+	 * ```
+	 */
+	function bindElementEvents(el: Highcharts.SVGElement, handlers: Highcharts.ElementEvents, boundEvents: Highcharts.BoundEvent[] = []) {
```

Comment:
Linter should guard a reasonable line length. If 80 is too rough, 120 shouldn't be exceeded for sure. Comments or strings could have exception only if it helps with readability.